### PR TITLE
Data-pipeline Ansible modification according to ubuntu-20.04

### PR DESCRIPTION
**Pull request by sivakarthikeyanoptit**
Data-pipeline Ansible modification according to ubuntu-20.04

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Tested in shikshalokam dev env
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/roles/ml-analytics-service/tasks/main.yml [hunk 1]
```diff
@@ -21,9 +21,8 @@
       - openjdk-8-jdk
       - software-properties-common
       - python3-pip
-      - python3.6-dev
-      - python3.6-venv
-      - python-virtualenv
+      - python3.8-venv
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/tasks/main.yml:24]: why are we specifying 3.8? won't 3-venv work?

**Review by rjshrjndrn — CHANGES_REQUESTED** [review]

File: ansible/roles/ml-analytics-service/tasks/main.yml (hunk 1 above)
Comment (sivakarthikeyanoptit) [reply ansible/roles/ml-analytics-service/tasks/main.yml]: Changed @rjshrjndrn

File: ansible/roles/ml-analytics-service/tasks/main.yml [hunk 2]
```diff
@@ -42,9 +41,9 @@
     - "{{ WORKDIR }}/faust_as_service"
 
 - name: Change user and create working directory under opt dir and install python virtual environment
-  shell: "cd {{ WORKDIR }} && virtualenv --python=python3.6 spark_venv"
-  become: yes
-  become_user: "{{ USER }}"
+  shell: "cd {{ WORKDIR }} && virtualenv --python=python3.8 spark_venv"
+  become: true
+  become_user: root
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/tasks/main.yml:46]: When you do become: yes it's root user. You don't have to specify that.

**Review by rjshrjndrn — CHANGES_REQUESTED** [review]

File: ansible/roles/ml-analytics-service/tasks/main.yml (hunk 2 above)
Comment (sivakarthikeyanoptit) [reply ansible/roles/ml-analytics-service/tasks/main.yml:46]: @rjshrjndrn i will remove that, then

Comment (sivakarthikeyanoptit) [reply ansible/roles/ml-analytics-service/tasks/main.yml]: done

File: ansible/roles/ml-analytics-service/defaults/main.yml [hunk 3]
```diff
@@ -4,4 +4,4 @@ BASEPATH: "/opt/sparkjobs" # Git Clone Path **Mandatory
 USER: "data-pipeline"
 WORKDIR: "/opt/sparkjobs"
 SYSTEMDDIR: "/lib/systemd/system"
-ml_analytics_version: "release-4.0.0"
+ml_analytics_version: "release-4.1.0"
```
Comment (rjshrjndrn) [inline ansible/roles/ml-analytics-service/defaults/main.yml:7]: Why do we need this variable here? It's supposed to be given from Jenkins job to clone the repo right?

**Review by rjshrjndrn — CHANGES_REQUESTED** [review]

File: ansible/roles/ml-analytics-service/defaults/main.yml (hunk 3 above)
Comment (rjshrjndrn) [reply ansible/roles/ml-analytics-service/defaults/main.yml:7]: Todo: chanbmge this

**Review by rjshrjndrn — APPROVED** [review]

**Pull request merged**